### ISSUE TITLE
link-to assertion fail: property modified twice in a single render

### ISSUE THREAD

**Issue by HenryVonfire** [ember-engines/ember-engines#667]
`ember-source`: from `3.10.0` to `3.10.2` (I haven't tried with other versions)
`ember-engines`: from `0.7.2` to `0.8.2`

Linking a property from a component or a controller to the `@route` of a `<LinkTo>` tag throws an assertion failed:

> Assertion Failed: You modified "url" twice on [object Object] in a single render. It was rendered in "component:link-to" and modified in "component:link-to". This was unreliable and slow in Ember 1.x and is no longer supported. See https://github.com/emberjs/ember.js/issues/13948 for more details.

This wasn't an issue in prior versions of `ember-engines`.

Way to reproduce the error:
- create a new project
- add ember-engines 0.7.2
- create an in-repo-engine
- create an application controller inside the engine
- create a string property with value `index` (for example test)
- add a `link-to` component to the application.hbs (angle brackets or not)
- add `this.test` as route for the `link-to` component
- run the application

When visiting the application route of the engine, it will throw the exception. I don't know if it can be relevant, but logging the property passed to the `link-to` component in the devtools, I saw that instead of being just the string `index`, `link-to` transforms it into a getter/setter.

**Comment by HenryVonfire** [comment]
I've tried to research a bit more about this issue and the problem seems to be that, in `didReceiveAttrs` of `link-to-component` component, the property `route` is set twice, one in [this component](https://github.com/ember-engines/ember-engines/blob/master/addon/components/link-to-component.js#L52) and another one in the [parent `link-to`](https://github.com/emberjs/ember.js/blob/37a19428dc59f0d0a7bb16c9c333b24ac262d81c/packages/%40ember/-internals/glimmer/lib/components/link-to.ts#L898) component.

Since the link-to component of `ember-engines` extends from the one provided by ember, when calling `this._super` inside `didReceiveAttrs` the `route` property is set twice.

Anyone has an idea of how to fix this? I tried to use `scheduleOnce` but that doesn't add the engine name in time for the route :disappointed: 

Another easy way to test this assertion error is by using the dummy app. Setting a property [`post` controller](https://github.com/ember-engines/ember-engines/blob/master/tests/dummy/lib/ember-blog/addon/controllers/post.js) of the `ember-blog` engine with the name of one of the routes used in the [`post` template](https://github.com/ember-engines/ember-engines/blob/master/tests/dummy/lib/ember-blog/addon/templates/post.hbs) will break almost all the tests for the `link-to` component.

**Comment by buschtoens** [comment]
Can you try submitting a PR with a failing test case?

**Comment by HenryVonfire** [comment]
Done, you can find it here #668

**Comment by thec0keman** [comment]
Does this need to be bumped up to the main ember repo?  Doing multiple sets of a property in `didReceiveAttrs` shouldn't throw an error like this, so it seems likely there is an undesirable coupling or side-effect here?

**Comment by thec0keman** [comment]
Looking into [ember-router-helpers](https://github.com/rwjblue/ember-router-helpers) as a possible workaround... comes with a couple of caveats though:

1) That addon does not seem to support engines, so you have to use fully formed paths (and they generate full-app page loads)

2) Our engines have a lot of `link-to`'s.... so this will be tedious.

The interesting thing is I think this is mostly tied to passing a bound attribute into the path property.  So for example, we set up a simple helper `dup` to wrap the value:

```
= lin-to (dup myObject.path)
```

Then everything works.  So it seems like the major issue here is when you pass bound objects into `link-to`, and ember-engines tries to prefix the engine name.

**Comment by mydea** [comment]
Just noting this here, we also ran into this and were able to workaround this by passing the argument as readonly:

```hbs
<LinkTo @route={{readonly this.dynamicRouteName}}>
```

It's not ideal, but at least it works!

**Comment by HenryVonfire** [comment]
`readonly` seems a better alternative in my opinion, since `unbound` is considered a legacy API and could be deprecated sooner or later.

**Comment by elwayman02** [comment]
I believe this is fixed now with #735 @rwjblue @HenryVonfire
